crnlib/crn_miniz.cpp

```diff
@@ -42,7 +42,7 @@ typedef unsigned char mz_validate_uint64[sizeof(mz_uint64) == 8 ? 1 : -1];
 #ifdef _MSC_VER
 #define MZ_FORCEINLINE __forceinline
 #elif defined(__GNUC__)
-#define MZ_FORCEINLINE __attribute__((__always_inline__))
+#define MZ_FORCEINLINE __attribute__((__always_inline__)) inline

```

Comment:
`inline` in current language standards no longer has any relation with inlining. It means that when linking there is allowed to be more than one definition of the function

Reply:
The warning:

```
warning: ‘always_inline’ function might not be inlinable [-Wattributes]
```

Reply:
Well this is a bunch of crap written to micro-optimize some compiler 20 years ago. It would be better to just nuke the always_inline thing than interfere with the linkage of the function.

Reply:
Yes, but this is a third-party lib, so I prefer the less-intrusive solution for now.

We may want to update miniz and lzma at some point, the same way I updated stb_image, but for now all I want is to do the less intrusive changes possibles while fixing the obvious things.

Reply:
I was going to complain that changing the function linkage is not the least intrusive thing, but it turns out that all MZ_FORCEINLINE functions are declared with `static` as well. When both static and inline linkage keywords are present, static takes priority, so their linkage won't change. So I guess it's OK.